From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dreeam <61569423+Dreeam-qwq@users.noreply.github.com>
Date: Fri, 18 Nov 2022 23:26:16 -0500
Subject: [PATCH] Remove UseItemOnPacket Too Far check

This Check is added in 1.17.x -> 1.18.x that updated by Mojang.
By removing this check, it gives ability for hackers to use some modules of hack clients.

diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 4b59db324e71846d233e43bafcdae658bcc70328..eba90724a6e5e8a93d775d3e0703a0b37a77e7e5 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2003,8 +2003,13 @@ public class ServerGamePacketListenerImpl
                 BlockPos blockPos = hitResult.getBlockPos();
                 if (this.player.canInteractWithBlock(blockPos, 1.0)) {
                     Vec3 vec3 = location.subtract(Vec3.atCenterOf(blockPos));
-                    double d = 1.0000001;
-                    if (Math.abs(vec3.x()) < 1.0000001 && Math.abs(vec3.y()) < 1.0000001 && Math.abs(vec3.z()) < 1.0000001) {
+                    // Leaf start - Remove UseItemOnPacket Too Far check
+                    //double d = 1.0000001;
+                    final double maxDistance = org.dreeam.leaf.config.modules.gameplay.ConfigurableMaxUseItemDistance.maxUseItemDistance;
+                    if (maxDistance <= 0
+                        || (Math.abs(vec3.x()) < maxDistance && Math.abs(vec3.y()) < maxDistance && Math.abs(vec3.z()) < maxDistance)
+                    ) {
+                        // Leaf end - Remove UseItemOnPacket Too Far check
                         Direction direction = hitResult.getDirection();
                         this.player.resetLastActionTime();
                         int maxY = this.player.level().getMaxY();
